crates/bifrost/src/loglets/local_loglet/provider.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::{hash_map, HashMap};
+use std::path::Path;
+use std::sync::{Arc, OnceLock};
+
+use async_trait::async_trait;
+use tokio::sync::Mutex as AsyncMutex;
+use tracing::debug;
+
+use super::log_store::RocksDbLogStore;
+use super::log_store_writer::RocksDbLogWriterHandle;
+use super::{LocalLoglet, Options};
+use crate::loglet::{Loglet, LogletOffset, LogletProvider};
+use crate::loglets::local_loglet::log_store_writer::WriterOptions;
+use crate::{Error, LogletParams};
+
+#[derive(Debug)]
+pub struct LocalLogletProvider {
+    log_store: RocksDbLogStore,
+    active_loglets: AsyncMutex<HashMap<String, Arc<LocalLoglet>>>,
+    log_writer: OnceLock<RocksDbLogWriterHandle>,
+    opts: Options,
+}
+
+impl LocalLogletProvider {
+    pub fn new(storage_path: &Path, raw_options: &serde_json::Value) -> Arc<Self> {
+        let opts =
+            serde_json::from_value(raw_options.clone()).expect("to be able to deserialize options");
+        // todo: implement loglet loading error handling
```

Comment:
Once we have error handling for loglet providers, could we move the log_writer creation to the `new` function? Then we wouldn't need the `OnceLock`.

Reply:
I think the answer would be no since we need to start the log writer to obtain the handle which we set in the `OnceLock`.